connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConfig.java

```diff
@@ -239,6 +257,63 @@ protected static ConfigDef baseConfigDef() {
                         Importance.LOW, HEADER_CONVERTER_CLASS_DOC);
     }
 
+    private void logInternalConverterDeprecationWarnings(Map<String, String> props) {
+        String[] deprecatedConfigs = new String[] {
+            INTERNAL_KEY_CONVERTER_CLASS_CONFIG,
+            INTERNAL_VALUE_CONVERTER_CLASS_CONFIG
+        };
+        for (String config : deprecatedConfigs) {
+            if (props.containsKey(config)) {
+                if (INTERNAL_CONVERTER_DEFAULT.equals(getClass(config))) {
+                    log.info(
+                        "Configuration {} is deprecated and will be removed in an upcoming release. "
+                            + "This value matches the current default, so it can be removed without concern.",
+                        config
+                    );
+                } else {
+                    log.warn(
+                        "Configuration {} is deprecated and will be removed in an upcoming release. "
+                            + "The current value '{}' does NOT match the default of {}, and behavior may change when removed.",
+                        config,
+                        getClass(config).getCanonicalName(),
+                        INTERNAL_CONVERTER_DEFAULT.getCanonicalName()
+                    );
+                }
+            }
+
+            for (Map.Entry<String, String> propEntry : props.entrySet()) {
+                String prop = propEntry.getKey();
+                if (prop.startsWith(config + ".")) {
+                    if ((config + "." + JsonConverterConfig.SCHEMAS_ENABLE_CONFIG).equals(prop)) {
+                        if (Boolean.FALSE.toString().equals(propEntry.getValue())) {
+                            log.info(
+                                "Configuration {} (along with all configuration for {}) is deprecated and will be removed in an upcoming release. "
+                                    + "This value matches the current default, so it can be removed without concern.",
+                                prop,
+                                config
+                            );
+                        } else {
+                            log.warn(
+                                "Configuration {} (along with all configuration for {}) is deprecated and will be removed in an upcoming release. "
+                                    + "The current value '{}' does NOT match the default of false, and behavior may change when removed.",
+                                prop,
+                                config,
+                                get(prop)
+                            );
+                        }
+                    } else {
+                        log.warn(
+                            "Configuration property '{}' (along with all configuration for {}) is deprecated and will be removed in an upcoming release. "
+                                + "The current value has no default, and behavior may change when removed.",
```

Comment:
We actually don't know what the default is for these "extra" properties.

So this logic exploded beyond what I'd hoped, and while I still think it's worthwhile to point out the case when specific properties can safely be removed, I'd be interested in whether other people find that's worth the effort.

@C0urante, is there any way of simplifying the logic? What about creating a helper method that, given the key, value, and possibly null default value, just logs the message. Then, the `logInternalConverterDeprecationWarnings` method could call this method for the various cases. Seems like it might reduce the slight repetitive logic but more importantly make it easier to see the logic that `logInternalConverterDeprecationWarnings` is using. WDYT?

Reply:
The complicated logic here is focused on catching properties prefixed with `internal.(key|value).converter`, though. A helper method for key/value pairs would simplify things but only because it wouldn't handle actually handle that.

It still seems important to catch properties that appear to be aimed at the internal converters. I'll try to simplify the logic without sacrificing that.